noxfile.py

```diff
@@ -26,3 +30,11 @@ def run_template_generation(session):
     session.install("setuptools", silent=False)
     session.install("-e", ".[dev]", silent=False)
     session.run("pytest", "tests")
+
+@nox.session(name="dev")
+def set_dev(session):
+    """Install pybamm-cookiecutter in editable mode"""
+    session.install("virtualenv")
+    session.run("virtualenv", os.fsdecode(VENV_DIR), silent=True)
+    python = os.fsdecode(VENV_DIR.joinpath("bin/python"))
+    session.run(python, "-m", "uv", "pip", "install", "-e", ".[dev]")
```

Comment:
I tried this locally and it isn't working – `uv` doesn't seem to properly install dependencies into a `virtualenv`-created environment. I think it needs itself to create the virtual environment, i.e., using `uv venv`. I don't wish to hold the PR, so maybe we can use the original, non-`uv` solution and fix this later?

Reply:
Strange, it works for me when I use nox outside the `virtualenv` and it detects the `venv` folder in the working directory and installs the project inside it, perhaps we should look at this later when we are done with model entry points.

Reply:
I don't want to hold this PR either, but would it be fine to test this session on CI as well?

Reply:
Resolving this, at the same time RFD on #21